spec/models/king_spec.rb

```diff
@@ -2,82 +2,175 @@
+      expect(king.valid_move?(2,0)).to eq false
+    end
+
+    it 'should return false if attempting to castle queenside and knight present' do
+      game = Game.create(white_user_id: 0, black_user_id: 1)
+      king = King.create(x_position: 4, y_position: 0, game_id: game.id)
+      rook = Rook.create(x_position: 0, y_position: 0, game_id: game.id)
+      knight = Knight.create(x_position: 1, y_position:0, game_id: game.id)
+      expect(king.valid_move?(2,0)).to eq false
+    end
+
+    it 'should return false if attempting to castle queenside and queen present' do
+      game = Game.create(white_user_id: 0, black_user_id: 1)
+      king = King.create(x_position: 4, y_position: 0, game_id: game.id)
+      rook = Rook.create(x_position: 0, y_position: 0, game_id: game.id)
+      queen = Queen.create(x_position: 3, y_position:0, game_id: game.id)
+      expect(king.valid_move?(2,0)).to eq false
+    end    
+
+    it 'should return false if  attempting to castle kingside and 7 rook has been moved' do
+      game = Game.create(white_user_id: 0, black_user_id: 1)
+      king = King.create(x_position: 4, y_position: 0, game_id: game.id)
+      rook = Rook.create(x_position: 7, y_position: 0, game_id: game.id)
+      rook.move_to!(6,0)
+      rook.move_to!(7,0)
+      expect(king.valid_move?(6,0)).to eq false
+    end
+
+    it 'should return false if attempting to castle queenside and 0 rook has been moved' do
+      game = Game.create(white_user_id: 0, black_user_id: 1)
+      king = King.create(x_position: 4, y_position: 0, game_id: game.id)
+      rook = Rook.create(x_position: 0, y_position: 0, game_id: game.id)
+      rook.move_to!(1,0)
+      rook.move_to!(0,0)
+      expect(king.valid_move?(6,0)).to eq false
+    end
+
+    it 'should return false if attempting to castle and king has been moved' do
+      game = Game.create(white_user_id: 0, black_user_id: 1)
+      king = King.create(x_position: 4, y_position: 0, game_id: game.id)
+      rook = Rook.create(x_position: 7, y_position: 0, game_id: game.id)
+      king.move_to!(3,0)
+      king.move_to!(4,0)
+      expect(king.valid_move?(6,0)).to eq false
+    end
+
+    it 'should return false if attempting to move 2 squares south' do
+      game = Game.create(white_user_id: 0, black_user_id: 1)
+      king = King.create(x_position: 4, y_position: 0, game_id: game.id)
+      expect(king.valid_move?(4,2)).to eq false
+    end
+
+    it 'should return false if attempting to move 2 squares north' do
+      game = Game.create(white_user_id: 0, black_user_id: 1)
+      king = King.create(x_position: 4, y_position: 7, game_id: game.id)
+      expect(king.valid_move?(4,5)).to eq false
+    end
+
     it 'should return true if moving one square east' do
       game = Game.create(white_user_id: 0, black_user_id: 1)
       king = King.create(x_position: 4, y_position: 0, game_id: game.id)
-      expect(king.valid_move?(5,0)).to eq true
+      expect(king.normal_valid_move?(5,0)).to eq true
```

Comment:
The tests should still reference `valid_move?` not `normal_valid_move?` as that's what the rest of the app will be using.  In this case, `normal_valid_move?` is an "internal" method, part of how king's implement `valid_move?`.